pygeoapi/provider/base.py

```diff
@@ -51,6 +51,7 @@ def __init__(self, provider_def):
         except KeyError:
             raise RuntimeError('name/type/data are required')
 
+        self.metadata = provider_def.get('metadata')
```

Comment:
Shouldn't this line be more defensive, i.e. provide a default of `None` like the other fields?